File: src/main/java/chat/sprchat/state/ConnectedClient.java
package chat.sprchat.state;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import org.java_websocket.WebSocket;
import org.java_websocket.handshake.ClientHandshake;

@Getter @Setter
public class ConnectedClient
{
    private WebSocket socket;
    private ClientHandshake clientHandshake;
    private String name;

    public ConnectedClient(WebSocket socket, ClientHandshake clientHandshake)
    {
        this.socket = socket;
        this.clientHandshake = clientHandshake;
    }
}

File: src/main/java/chat/sprchat/state/ConnectedClient.kt
package chat.sprchat.state

import org.java_websocket.WebSocket
import org.java_websocket.handshake.ClientHandshake

data class ConnectedClient(
        private val socket: WebSocket,
        private val clientHandshake: ClientHandshake,
){
    private lateinit var name: String
    fun getSocket(): WebSocket{
        return socket
    }
    fun getClientHandshake(): ClientHandshake{
        return clientHandshake
    }
    fun getName(): String{
        return name
    }
    fun setName(_name: String){
        name = _name
    }
}